shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs

```diff
@@ -1373,3 +1374,38 @@ fn test_cassandra_protect_transform_aws() {
     functions::test(&shotover_connection);
     protect::test(&shotover_connection, &direct_connection);
 }
+
+#[test]
+#[serial]
+fn test_cassandra_peers_rewrite() {
+    let _docker_compose =
+        DockerCompose::new("tests/test-topologies/cassandra-peers-rewrite/docker-compose.yml")
+            .wait_for_n_t("Startup complete", 2, 90);
+
+    let shotover_manager = ShotoverManager::from_topology_file(
+        "tests/test-topologies/cassandra-peers-rewrite/topology.yaml",
+    );
+
+    let normal_connection = shotover_manager.cassandra_connection("127.0.0.1", 9043);
+    let rewrite_port_connection = shotover_manager.cassandra_connection("127.0.0.1", 9044);
+    let emulate_single_node_port_connection =
+        shotover_manager.cassandra_connection("127.0.0.1", 9045);
+
+    assert_query_result(
+        &normal_connection,
+        "SELECT native_port FROM system.peers_v2;",
+        &[&[ResultValue::Int(9042)]],
+    );
+
+    assert_query_result(
+        &rewrite_port_connection,
+        "SELECT native_port FROM system.peers_v2;",
+        &[&[ResultValue::Int(9044)]],
+    );
```

Comment:
We should also have a case with multiple columns, maybe `SELECT * FROM system.peers_v2;`, to ensure that our logic for modifying just the native_port is correct.

Reply:
I've added a couple extra columns, I didn't want to use `*` as the `tokens` column would make the expected results very large.

Reply:
Thats fine, can we also keep the original test case though?
1 returned value is a nice edge case to test.

Reply:
Sure, I'll add it back.